src/install-peerdeps.js

```diff
@@ -2,8 +2,6 @@
 import "@babel/polyfill";
 
 import fs from "fs";
-import request from "request-promise-native";
-import HttpsProxyAgent from "https-proxy-agent";
```

Comment:
i suspect removing this dep will address #80 (albeit only for a v3) by restoring support for older nodes. It'd be good to add to the test matrix, and the "engines" field, as far back as this PR works.

Reply:
Babel 7 dropped support for Node 5 and below, so I wouldn't attempt trying to bring back Node support for the ancient versions reported in #80. If you're running a version of Node that old, you're probably having to pin a lot of dependencies.

Reply:
In addition, `pnpm` dropped support for Node 8 in v4.0.0:
https://github.com/pnpm/pnpm/releases/tag/v4.0.0

So I'm going to remove Node 8 from the test matrix and flag this as another breaking change.

If there's a desire to provide a semver-correct fix for old versions of Node, then I suggest making a maintenance branch and doing the work there. But I don't have the capacity to go digging into the history of how old versions of this package and old versions of Node worked.

Reply:
Babel’s supported version doesn’t matter, we can build in new node and test against old nodes.

Please do not remove node 8 from the matrix.

Reply:
@ljharb the Node 8 tests are failing, (and also failing on your master branch) because newer versions of `pnpm` don't work on Node 8.. I don't have `pnpm` and not going to try to figure out how to get an old version of it working on Travis. Your master branch is failing and not working for Node 8. The intent of this PR is not to reestablish backwards support for old versions of Node and various libraries. The intent of this PR is to replace the direct API calls with those already afforded by the package manager, so that all the proxy and repository logic can be offloaded to the package manager where it is better handled.

Reply:
in that case, we should pin to a version of pnpm that does. That the tests fail due to pnpm doesn't mean that *this* package is failing in node 8.

Reply:
So go figure that out and add it to master. I'll happily merge that into this branch.